node-backend/controllers/v1/s3SignedUrlController.js

```diff
@@ -0,0 +1,66 @@
+var AWS = require("aws-sdk");
+const log4js = require("log4js");
+const qs = require("qs"); //eslint-disable-line node/no-extraneous-require
+
+const jsonwebtoken = require("../../jwtTokenValidation/jwtValidation");
+const utility = require("../../utils/utility");
+const validateSchema = require("./validationSchema/s3SignedUrlValidation");
+require("../../config/loggerConfig");
+const logger = log4js.getLogger();
+
+module.exports.getSignedUrl = async (req, res) => {
+  const authHeader = req.headers["authorization"];
+  const decode = await jsonwebtoken.getData(authHeader);
+  const schema = validateSchema.s3SignedUrl();
+  let obj = qs.parse(req.query);
+  if (obj.type == "profile" || obj.type == "core_value") {
+    schema
+      .validate(obj, { abortEarly: false })
+      .then(async () => {
+        const credentials = {
+          accessKeyId: process.env.S3_BUCKET_ACCESS_KEY,
+          secretAccessKey: process.env.S3_BUCKET_SECRET_KEY,
+        };
+        AWS.config.update({
+          credentials: credentials,
+          region: process.env.REGION,
+        });
+        const s3 = new AWS.S3();
+        const presignedURL = await s3.getSignedUrl("putObject", {
+          ContentType: process.env.S3_BUCKET_CONTENT_TYPE,
```

Comment:
I guess this is a file extension. If it is so, then please accept file type as a parameter from user